### DISCUSSION TITLE
add gemini pro support

### DISCUSSION THREAD

**Discussion by theurs** [tbckr/sgpt#232] [Ideas]
It's free and very easy to use
There are examples with curl

```
curl https://generativelanguage.googleapis.com/v1beta/models/gemini-pro:generateContent?key=$API_KEY \
    -H 'Content-Type: application/json' \
    -X POST \
    -d '{
      "contents": [{
        "parts":[{
          "text": "Write a story about a magic backpack."}]}]}' 2> /dev/null
```

**Comment by tbckr** [comment]
Thank you for your request! I think this is a cool feature too. I have it on the roadmap: #173

I will look into it in the next few weeks, but do not know when it will be implemented (it needs some refactoring).

**Comment by johnd0e** [comment]
@theurs 
You can use `sgpt` with Google Gemini API with help of this project: https://github.com/PublicAffairs/openai-gemini
